substratevm/src/com.oracle.svm.reflect/src/com/oracle/svm/reflect/serialize/SerializationSupport.java

```diff
@@ -0,0 +1,222 @@
+            } else {
+                System.out.println(errorMsg);
+                System.out.println("Compilation will continue because " + option +
+                                " was set. But the program may behave unexpectedly at runtime.");
+            }
+        }
+    }
+
+    @Platforms({Platform.HOSTED_ONLY.class})
+    private void reportMultiClassLoaderIssue(FeatureImpl.DuringSetupAccessImpl access) {
+        StringBuilder sb = new StringBuilder();
+        serializationTargetAndChecksums.forEach((targetClass, definitions) -> {
+            int size = definitions.size();
+            if (size > 1) {
+                sb.append("Suspicious multiple-classloader usage is detected from serialization configurations:\n");
+                sb.append("There are " + size + " SerializationConstructorAccessor classes have been defined for the same serialization target class:\n\n");
+                int i = 0;
+                while (i < size) {
+                    sb.append("(").append((i + 1)).append(")");
+                    sb.append(definitions.get(i).toString());
+                    sb.append("\n");
+                    i++;
+                }
+            }
+        });
+        String exceptionsMsg = sb.toString();
+        // Checkstyle: stop
+        if (exceptionsMsg.length() > 0) {
+            String option = SubstrateOptionsParser.commandArgument(NativeImageOptions.ReportUnsupportedElementsAtRuntime, "+");
+            if (!NativeImageOptions.ReportUnsupportedElementsAtRuntime.getValue()) {
+                BigBang bb = access.getBigBang();
+                bb.getUnsupportedFeatures().addMessage(MULTIPLE_CHECKSUMS, null,
+                                exceptionsMsg + "\n" + "To allow continuing compilation with above unsupported features, set " + option);
+            } else {
+                System.out.println(exceptionsMsg);
+                System.out.println("Compilation will continue because " + option +
+                                " was set. But the program may behave unexpectedly at runtime.");
+            }
+        }
+        // Checkstyle: resume
+    }
+
+    @Platforms({Platform.HOSTED_ONLY.class})
+    @Override
+    public Class<?> addSerializationConstructorAccessorClass(Class<?> serializationTargetClass, Long configuredChecksum) {
+        if (serializationTargetClass.isArray() || serializationTargetClass.isEnum()) {
+            return null;
+        }
+        // Don't generate SerializationConstructorAccessor class for Externalizable case
+        if (Externalizable.class.isAssignableFrom(serializationTargetClass)) {
+            try {
+                Method getExternalizableConstructor = ObjectStreamClass.class.getDeclaredMethod("getExternalizableConstructor", Class.class);
+                getExternalizableConstructor.setAccessible(true);
+                Constructor<?> c = (Constructor<?>) getExternalizableConstructor.invoke(null, serializationTargetClass);
```

Comment:
Please use `ReflectionUtil` here and in the rest of the code.

Reply:
fixed

Reply:
fixed

Reply:
fixed